code/python/jeff/python/unit_converter_4.py

```diff
@@ -0,0 +1,34 @@
+distance_unit = {
+    'ft': 0.3048,
+    'me': 1,
+    'mi': 1609.34,
+    'km': 1000, 
+}
+
+distance_unit_2 = {
+    'ft': 0.3048,
+    'me': 1,
+    'mi': 1609.34,
+    'km': 1000, 
+}
+
+def unit_converter():
+    distance_input = int(input('\nPlease enter the distance: \n'))
+    unit_type = input('\nPlease enter the type of unit: ft for foot, mi for mile,\nme for meter, km for kilometer, yd for yard, in for inch : ')
+    subtotal = distance_unit[unit_type] * distance_input
+    distance_output = input('\nPlease enter the output units: \n')
+    total = subtotal * distance_unit_2[distance_output]
+    #input('n\Please enter the distance in feet') 
+    return f"Your conversion is equal to {total} {distance_output} " 
+
+distance_unit.update({'yd': 0.9144 , 'in': 0.0254})
```

Comment:
Right here you updated a dictionary correctly, but you updated the wrong dictionary. If you look at your code you reference distance_unit_2 in the function unit_converter(). Just change line 24 from "distance_unit" into "distance_unit_2.update....." and you're good as gold. Greate job!